xbmc/addons/AddonManager.h

```diff
@@ -137,10 +137,12 @@ namespace ADDON
      * @brief Disable addons in given list.
      *
      * @param[in] incompatible List of incompatible addon infos
+     * @param[in] includedependencies if dependency addons are to be included in the list of
+     * returned disabled addon names
      * @return list of all addon **names** that were disabled
      */
     std::vector<std::string> DisableIncompatibleAddons(
-        const std::vector<AddonInfoPtr>& incompatible);
+        const std::vector<AddonInfoPtr>& incompatible, bool includedependencies);
```

Comment:
Why not a new utility function that returns an addon list excluding dependencies instead of overloading this functions which feels a bit hacky?

So like: `std::vector<std::string> GetAddonListWithoutDependencies(const std::vector<AddonInfoPtr>& addonInfos);`

Reply:
Yes is really hacky, the call "DisableIncompatibleAddons" does the work for it and his list should always be complete of disabled ones.

To use another function is more clean.